drivers/entropy/entropy_ateccx08.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2024 Vogl Electronic GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT microchip_ateccx08_entropy
+
+#include <errno.h>
+#include <string.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/entropy.h>
+#include <zephyr/drivers/eeprom/ateccx08.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(entropy_ateccx08, CONFIG_ENTROPY_LOG_LEVEL);
+
+struct entropy_ateccx08_config {
+	const struct device *parent;
+};
+
+struct entropy_ateccx08_data {
+	uint8_t random_buffer[32];
+	size_t buffer_pos;
+};
+
+static int entropy_ateccx08_get_entropy(const struct device *dev, uint8_t *buffer, uint16_t length)
+{
+	const struct entropy_ateccx08_config *config = dev->config;
+	struct entropy_ateccx08_data *dev_data = dev->data;
+
+	int ret = 0;
+
+	if (atecc_is_locked_config(config->parent) == false) {
+		LOG_ERR("Config not locked, no random data available.");
+		return -EPERM;
+	}
+
+	if (dev_data->buffer_pos > 0) {
+		size_t to_copy = MIN(length, dev_data->buffer_pos);
+		uint8_t *pos = dev_data->random_buffer + sizeof(dev_data->random_buffer) -
+			       dev_data->buffer_pos;
+
+		memcpy(buffer, pos, to_copy);
+		buffer += to_copy;
+		length -= to_copy;
+		dev_data->buffer_pos -= to_copy;
+	}
+
+	while (length > 0) {
+		size_t to_copy;
+
+		ret = atecc_random(config->parent, dev_data->random_buffer, true);
+		if (ret < 0) {
+			return ret;
+		}
+
+		to_copy = MIN(length, sizeof(dev_data->random_buffer));
+
+		memcpy(buffer, dev_data->random_buffer, to_copy);
+		buffer += to_copy;
+		length -= to_copy;
+		dev_data->buffer_pos = sizeof(dev_data->random_buffer) - to_copy;
```

Comment:
this variable is misleading. This is not the buffer position but how much space is available in the buffer, right ?

That's why in the previous block you have
```
uint8_t *pos = dev_data->random_buffer + sizeof(dev_data->random_buffer) -  dev_data->buffer_pos;
```
instead of:
```
uint8_t *pos = dev_data->random_buffer + dev_data->buffer_pos;
````